### PR TITLE
verifier

### PR DESCRIPTION
**Pull request by laudiacay**

### CODE REVIEW

**Comment by laudiacay** [comment]
@0xJepsen @Autoparallel small refactor afoot, ignore me (but- fyi- don't touch that verifier test too much

**Comment by laudiacay** [comment]
bigger refactor than expected, everything but the test compiles. test does not compile and i'm getting too tired to fix it. lot of warnings but i don't care about those yet...

File: barustenberg/src/ecc/curves/bn254_scalar_multiplication.rs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    let numerator = (-F::from(3_u32)).sqrt().unwrap() - F::from(1_u32);
+    two_inv * numerator
+    // constexpr field two_inv = field(2).invert();
+    // constexpr field numerator = (-field(3)).sqrt() - field(1);
+    // constexpr field result = two_inv * numerator;
+    // return result;
+    //}
+}
+pub(crate) fn is_point_at_infinity(point: &G1Projective) -> bool {
+    point.x.is_zero() && point.y.is_one() && point.z.is_zero()
```
Comment (Autoparallel) [inline barustenberg/src/ecc/curves/bn254_scalar_multiplication.rs:30]: couldn't this just be 

`!point.x.is_zero() && !point.y.is_zero() && point.z.is.zero()`?

File: barustenberg/src/plonk/proof_system/commitment_scheme.rs [hunk 2]
```diff
@@ -12,81 +12,105 @@ use crate::transcript::{BarretenHasher, Transcript};
...
-pub(crate) trait CommitmentScheme<
-    Fq: Field,
-    Fr: Field + FftField,
-    G1Affine: AffineRepr,
-    H: BarretenHasher,
->
-{
+pub(crate) trait CommitmentScheme {
+    type Fq: Field + FftField;
+    type Fr: Field + FftField;
```
Comment (Autoparallel) [inline barustenberg/src/plonk/proof_system/commitment_scheme.rs:21]: These are the same type, don't we only need either `Fq` or `Fr`?

File: barustenberg/src/plonk/proof_system/commitment_scheme.rs [hunk 3]
```diff
@@ -12,81 +12,105 @@ use crate::transcript::{BarretenHasher, Transcript};
...
     );
 }
 
-#[derive(Default)]
-pub(crate) struct KateCommitmentScheme<H: BarretenHasher, S: Settings<H>> {
+#[derive(Default, Debug)]
+pub(crate) struct KateCommitmentScheme<
+    H: BarretenHasher,
+    Fq: Field + FftField,
+    Fr: Field + FftField,
```
Comment (Autoparallel) [inline barustenberg/src/plonk/proof_system/commitment_scheme.rs:82]: once again

**Review by Autoparallel — APPROVED** [review]
my eyes glazed over

File: barustenberg/src/ecc/curves/bn254_scalar_multiplication.rs [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+}
+pub(crate) fn generate_pippenger_point_table(
+    points: &mut [G1Affine],
+    table: &mut [G1Affine],
+    num_points: usize,
+) {
+    // calculate the cube root of unity
+    let beta = cube_root_of_unity::<ark_bn254::Fq>();
+
+    // iterate backwards, so that `points` and `table` can point to the same memory location
```
Comment (0xJepsen) [inline barustenberg/src/ecc/curves/bn254_scalar_multiplication.rs:69]: Nice

**Review by 0xJepsen — APPROVED** [review]

File: barustenberg/src/ecc/curves/bn254_scalar_multiplication.rs (hunk 1 above)
Comment (laudiacay) [reply barustenberg/src/ecc/curves/bn254_scalar_multiplication.rs:30]: it'd actually be point.x.is_zero() && !point.y.is_zero() && point.z.is.zero() but yes good catch mine was wrong

Comment (laudiacay) [reply barustenberg/src/ecc/curves/bn254_scalar_multiplication.rs:30]: fixed

**Pull request merged**